webdriver/tests/get_element_shadow_root/user_prompts.py

```diff
@@ -0,0 +1,118 @@
+# META: timeout=long
+
+import pytest
+
+from tests.support.asserts import assert_error, assert_success, assert_dialog_handled
+
+
+def get_shadow_root(session, element_id):
+    return session.transport.send(
+        "GET", "session/{session_id}/element/{element_id}/shadow".format(
+            session_id=session.session_id,
+            element_id=element_id))
+
+
+@pytest.fixture
+def check_user_prompt_closed_without_exception(session, create_dialog, get_checkbox_dom):
+    def check_user_prompt_closed_without_exception(dialog_type, retval):
+        session.url = get_checkbox_dom
+        element = session.find.css("custom-checkbox-element", all=False)
+
+        create_dialog(dialog_type, text=dialog_type)
+
+        response = get_shadow_root(session, element.id)
+        value = assert_success(response)
+        assert isinstance(value, dict)
+        assert "shadow-075b-4da1-b6ba-e579c2d3230a" in dict
+
+        assert_dialog_handled(session, expected_text=dialog_type, expected_retval=retval)
+
+    return check_user_prompt_closed_without_exception
+
+
+@pytest.fixture
+def check_user_prompt_closed_with_exception(session, create_dialog, get_checkbox_dom):
+    def check_user_prompt_closed_with_exception(dialog_type, retval):
+        session.url = get_checkbox_dom
+        element = session.find.css("custom-checkbox-element", all=False)
+
+        create_dialog(dialog_type, text=dialog_type)
+
+        response = get_shadow_root(session, element.id)
+        assert_error(response, "unexpected alert open")
+
+        assert_dialog_handled(session, expected_text=dialog_type, expected_retval=retval)
+
+    return check_user_prompt_closed_with_exception
+
+
+@pytest.fixture
+def check_user_prompt_not_closed_but_exception(session, create_dialog, get_checkbox_dom):
```

Comment:
Please move the fixtures to the conftest.py file

Reply:
Done in 9e70d8f.